### ISSUE TITLE
Timeout  doesn't work

### ISSUE THREAD

**Issue by jgcabral** [zytzagoo/smtp-validate-email#39]
I'm extending class and inside function constructor i set timeout = 250(ms) but i when execute the script delay more time. Often it happen with email address hotmail.

**Comment by zytzagoo** [comment]
Can you show where/how exactly you're doing that?
This class's constructor does not accept timeout as an option by default, so hard to tell what's going on in some custom code of yours...

Are you using `setConnectTimeout()`? Are we talking about connect timeout at all, or some other timeout?
[Unit tests](https://github.com/zytzagoo/smtp-validate-email/blob/master/tests/Unit/ValidatorTest.php#L115) show that setting/getting the connect timeout works as expected. [Functional test](https://github.com/zytzagoo/smtp-validate-email/blob/master/tests/Functional/ValidatorTest.php#L24)  too.

**Comment by jgcabral** [comment]
The code is the following.
$validator = new EmailValidator($email, $sender);
$validator->setConnectTimeout(0.25);
but when i enter any address hotmail's, delay a lot of time.

**Comment by zytzagoo** [comment]
Connect timeout must (currently) be specified as integer seconds, not float (https://secure.php.net/stream_set_timeout is used), so that might be one of the reasons why its "not working", but we'll have to define what not working really means in this context... 

Do you have some debug logs/timestamps which show that the delay is actually happening at the connect phase?

Could it be that Hotmail is throttling you due to bad/spammy behaviour and/or bad/spammy ip reputation/history?

**Comment by zytzagoo** [comment]
Does setting the connect timeout to 1 (second) help as a workaround? Or does that also have no effect at all?

**Comment by jgcabral** [comment]
You're right, i think hotmail's servers are generating delay. I have tried setting 1 second but no success. I need to set 250 ms  for any email server. I have no logs to show you,

**Comment by zytzagoo** [comment]
Feature request 1: Maybe being able to set sub-second connectTimeout, but even if implemented, it does not help with server-generated delays/throttling.

Feature request 2: Detecting delays/throttling (and maybe dropping the connection in that case, and it being somewhat configurable)

These are not something I planned on adding myself (no need for it), however, if someone contributes the code and tests for it, I'll more than gladly review and merge.

**Comment by jgcabral** [comment]
Ok, thanks anyway.
